libs/auth-api-lib/src/lib/services/user-profile.service.ts

```diff
@@ -1,34 +1,164 @@
 import { Inject, Injectable } from '@nestjs/common'
 
-import type { Auth } from '@island.is/auth-nest-tools'
-import { EinstaklingarApi } from '@island.is/clients/national-registry-v2'
+import type { Auth, User } from '@island.is/auth-nest-tools'
+import { AuthMiddleware } from '@island.is/auth-nest-tools'
+import {
+  EinstaklingarApi,
+  Heimilisfang,
+} from '@island.is/clients/national-registry-v2'
 import type { EinstaklingarGetEinstaklingurRequest } from '@island.is/clients/national-registry-v2'
+import { GetCompanyApi } from '@island.is/clients/rsk/company-registry'
+import { UserProfileApi } from '@island.is/clients/user-profile'
 import { LOGGER_PROVIDER } from '@island.is/logging'
 import type { Logger } from '@island.is/logging'
 
-import { IndividuaInfoDTO } from '../entities/dto/individual-info.dto'
+import { UserProfileDTO } from '../entities/dto/user-profile.dto'
+import type { GenderValue } from '../entities/dto/user-profile.dto'
+import { AddressDTO } from '../entities/dto/address.dto'
 
 @Injectable()
 export class UserProfileService {
   constructor(
-    @Inject(EinstaklingarApi)
     private individualApi: EinstaklingarApi,
+    private userProfileApi: UserProfileApi,
+    private companyRegistryApi: GetCompanyApi,
     @Inject(LOGGER_PROVIDER)
     private logger: Logger,
   ) {}
 
-  async findIndividual(auth: Auth): Promise<IndividuaInfoDTO> {
-    try {
-      const result = await this.individualApi.einstaklingarGetEinstaklingur(<
-        EinstaklingarGetEinstaklingurRequest
-      >{
-        id: auth.nationalId,
-      })
-
-      return { name: result.nafn }
-    } catch (error) {
-      this.logger.error('Error in findIndividual', error)
-      throw error
+  userProfileApiWithAuth(auth: Auth) {
+    return this.userProfileApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  async getUserProfileClaims(auth: User): Promise<UserProfileDTO> {
+    const isCompany = !!auth.nationalId.match(/^[4-7]\d{9}$/)
```

Comment:
In another places we have been using the `isCompany` check from the `kennitala` package. Wondering if we should be consistent with this check.

Reply:
Good point.

I wan't this logic to support temporary ids which `kennitala` does not support until [they release 2.0](https://github.com/HermannBjorgvin/Kennitala/pull/23).